x-pack/plugins/snapshot_restore/public/application/components/restore_snapshot_form/steps/step_logistics/step_logistics.tsx

```diff
@@ -220,52 +284,35 @@ export const RestoreSnapshotStepLogistics: React.FunctionComponent<StepProps> =
-                                    indices: [],
-                                  });
-                                }}
-                              >
-                                <FormattedMessage
-                                  id="xpack.snapshotRestore.restoreForm.stepLogistics.deselectAllIndicesLink"
-                                  defaultMessage="Deselect all"
-                                />
-                              </EuiLink>
-                            ) : (
-                              <EuiLink
-                                onClick={() => {
-                                  // TODO: Change this to setIndicesOptions() when https://github.com/elastic/eui/issues/2071 is fixed
-                                  indicesOptions.forEach((option: EuiSelectableOption) => {
-                                    option.checked = 'on';
-                                  });
-                                  updateRestoreSettings({ indices: [...snapshotIndices] });
-                                  setCachedRestoreSettings({
-                                    ...cachedRestoreSettings,
-                                    indices: [...snapshotIndices],
-                                  });
-                                }}
-                              >
-                                <FormattedMessage
-                                  id="xpack.snapshotRestore.restoreForm.stepLogistics.selectAllIndicesLink"
-                                  defaultMessage="Select all"
-                                />
-                              </EuiLink>
-                            ),
+                      <DataStreamsAndIndicesListHelpText
+                        onSelectionChange={(selection) => {
+                          if (selection === 'all') {
+                            // TODO: Change this to setIndicesOptions() when https://github.com/elastic/eui/issues/2071 is fixed
+                            indicesAndDataStreamsOptions.forEach((option: EuiSelectableOption) => {
+                              option.checked = 'on';
+                            });
+                            updateRestoreSettings({
+                              indices: [...snapshotIndicesAndDataStreams],
+                            });
+                            setCachedRestoreSettings({
+                              ...cachedRestoreSettings,
+                              indices: [...snapshotIndicesAndDataStreams],
+                            });
+                          } else {
+                            // TODO: Change this to setIndicesOptions() when https://github.com/elastic/eui/issues/2071 is fixed
+                            indicesAndDataStreamsOptions.forEach((option: EuiSelectableOption) => {
+                              option.checked = undefined;
+                            });
+                            updateRestoreSettings({ indices: [] });
```

Comment:
Minor nit: there's a pattern to this logic that we could extract into a single helper function. We could name it in a way that clarifies the intention behind the code, if you think this improves things.

```ts
// Not entirely sure if the name of this function is accurate.
function updateAllRestoreSettings(state, indicesAndDataStreams) {
  // TODO: Change this to setIndicesOptions() when https://github.com/elastic/eui/issues/2071 is fixed
  indicesAndDataStreamsOptions.forEach((option: EuiSelectableOption) => {
    option.checked = state;
  });
  updateRestoreSettings({ indices: indicesAndDataStreams });
  setCachedRestoreSettings({
    ...cachedRestoreSettings,
    indices: indicesAndDataStreams,
  });
}
```

Reply:
I am going to hold off making this change for now, I suspect we will revisit this UI in the not too distant future and then we might benefit from having the handlers colocated.